pkg/systemd/generate/containers.go

```diff
@@ -339,6 +349,27 @@ func executeContainerTemplate(info *containerInfo, options entities.GenerateSyst
 			}
 		}
 
+		// Unless the user explicitly set a restart policy, check
+		// whether the container was created with a custom one and use
+		// it instead of the default.
+		if options.RestartPolicy == nil {
```

Comment:
related to above,  we could set `options.RestartPolicy` back to nil if we cannot validate it